### PR TITLE
add external_id and backend naive-implementation for contact_sql upload

### PR DESCRIPTION
**Pull request by schuyler1d**
this is a WIP to data from a data warehouse.  We will need to expose the frontend SQL form.  However, I think we should merge this in early because:
1. it adds external_id to the campaign_contact table which will probably be useful for Scott's actionkit_event connector
2. the backend does work, and does no harm.

### CODE REVIEW

File: src/components/CampaignContactsForm.jsx [hunk 1]
```diff
@@ -209,8 +209,10 @@ export default class CampaignContactsForm extends React.Component {
         Your upload file should be in CSV format with column headings in
         the first row. You must include <span className={css(styles.csvHeader)}>firstName</span>,
         <span className={css(styles.csvHeader)}>lastName</span>, and
-        <span className={css(styles.csvHeader)}>cell</span> columns. If you include a <span className={css(styles.csvHeader)}>zip</span> column,
+        <span className={css(styles.csvHeader)}>cell</span> columns.
+        If you include a <span className={css(styles.csvHeader)}>zip</span> column,
         we'll use the zip to guess the contact's timezone for enforcing texting hours.
+        An optional column to map the contact to a CRM is <span className={css(styles.csvHeader)}>external_id</span>
```
Comment (shakalee14) [inline src/components/CampaignContactsForm.jsx:214]: ok cool! so we would need to add this column.

Comment (schuyler1d) [reply src/components/CampaignContactsForm.jsx:214]: to the csv (and in the future as a column in the DB sql), yes, but then we can update/sync/send actions based on it

File: src/containers/AdminCampaignEdit.jsx [hunk 2]
```diff
@@ -145,7 +145,8 @@ class AdminCampaignEdit extends React.Component {
             cell: contact.cell,
             firstName: contact.firstName,
             lastName: contact.lastName,
-            zip: contact.zip
+            zip: contact.zip,
+            external_id: contact.zip
```
Comment (shakalee14) [inline src/containers/AdminCampaignEdit.jsx:151]: why is this contact.zip too?

File: src/workers/jobs.js [hunk 3]
```diff
@@ -60,6 +64,62 @@ export async function uploadContacts(job) {
   }
 }
 
+
```
Comment (shakalee14) [inline src/workers/jobs.js:67]: so we will have an input box, and then this function will pass it through line 79 as long as there is a datawarehouse connection, it starts with a select and doesnt start with a ;, cool!

File: src/workers/jobs.js [hunk 4]
```diff
@@ -60,6 +64,62 @@ export async function uploadContacts(job) {
...
+  if (!datawarehouse) {
+    log.error('No data warehouse connection, so cannot load contacts', job)
+    return
+  }
+
+  knexResult = await datawarehouse.raw(sql_query)
+  let fields = {}
+  let customFields = {}
+  const contactFields = {
+    first_name: 1,
```
Comment (shakalee14) [inline src/workers/jobs.js:83]: these 1s here are for `true`?

Comment (schuyler1d) [reply src/workers/jobs.js:83]: the value doesn't matter.  I just want to test whether it's in there.  I don't think es6 has sets, but i bet if i google it, they will.

File: src/containers/AdminCampaignEdit.jsx (hunk 2 above)
Comment (schuyler1d) [reply src/containers/AdminCampaignEdit.jsx:151]: the zip line is just modified because of adding the comma

**Pull request merged**